src/app/core/admin/user-management/admin-list-users.component.ts

```diff
@@ -159,11 +161,38 @@ export class AdminListUsersComponent extends AbstractPageableDataComponent<User>
 	}
 
 	exportUserData() {
-		console.error('Export User Data coming soon...');
+		console.error('Export of single user field is not yet supported.');
 	}
 
 	exportCurrentView() {
-		console.error('Export Current View coming soon...');
+		const viewColumns = Object.keys(this.columns)
+			.filter((key: string) => this.columns[key].show)
+			.map((key: string) => ({ key, title: this.columns[key].display }));
+
+		const rolesIndex = viewColumns.findIndex((pair: any) => pair.key === 'roles');
+
+		if (rolesIndex !== -1) {
+			viewColumns.splice(
+				rolesIndex,
+				1,
+				{ key: 'roles.user', title: 'User Role' },
+				{ key: 'roles.editor', title: 'Editor Role' },
+				{ key: 'roles.auditor', title: 'Auditor Role' },
+				{ key: 'roles.admin', title: 'Admin Role' }
+			);
+		}
+
+		this.exportConfigService
+			.postExportConfig('user', {
+				q: this.getQuery(),
+				s: this.search,
+				cols: viewColumns,
+				sort: this.pagingOptions.sortField,
+				dir: this.pagingOptions.sortDir
+			})
+			.subscribe((response: any) => {
+				window.open(`/api/admin/users/csv/${response._id}`);
```

Comment:
Interesting-- so it looks does essentially a post-redirect-get and defers the CSV generation to the server? If the client already has a given display window with data, is there a reason that we aren't generating the CSV in the client code?

I could definitely see us moving toward being able to export every pageable table to a downloaded file, and I'm trying to think of the maintenance lift to have to add a new endpoint to every call.

Reply:
I think the goal is to be able to export all pages of data. In that case, we need to go to the server to get all of that information.

Reply:
Seems reasonable.

Reply:
Thanks @tzfx ! I just added some spec coverage for this component and the export functionality.